### ISSUE TITLE
NullReferenceException when leaving a page after selecting a suggestion

### ISSUE THREAD

**Issue by Sim-github** [dotMorten/XamarinFormsControls#31]
Hello

I have the following exception on Android using V0.5.0
```
01-19 15:57:02.988 I/MonoDroid(25422): System.NullReferenceException: Object reference not set to an instance of an object.
01-19 15:57:02.988 I/MonoDroid(25422):   at dotMorten.Xamarin.Forms.AutoSuggestBoxRenderer.<UpdateItemsSource>b__14_0 (System.Object o) [0x00000] in E:\GitHub\dotMorten\XamarinFormsControls\AutoSuggestBox\dotMorten.Xamarin.Forms.AutoSuggestBox\AutoSuggestBoxRenderer.cs:233 
01-19 15:57:02.988 I/MonoDroid(25422):   at dotMorten.Xamarin.Forms.NativeAutoSuggestBox+SuggestCompleteAdapter+<>c__DisplayClass4_0.<UpdateList>b__0 (System.Object s) [0x00000] in E:\GitHub\dotMorten\XamarinFormsControls\AutoSuggestBox\dotMorten.Xamarin.Forms.AutoSuggestBox\NativeAutoSuggestBox.Android.cs:147 
01-19 15:57:02.988 I/MonoDroid(25422):   at System.Linq.Enumerable+SelectListIterator`2[TSource,TResult].ToArray () [0x00034] in <715c2ff6913942e6aa8535593b3ef35a>:0 
01-19 15:57:02.988 I/MonoDroid(25422):   at System.Linq.Enumerable.ToArray[TSource] (System.Collections.Generic.IEnumerable`1[T] source) [0x0001f] in <715c2ff6913942e6aa8535593b3ef35a>:0 
01-19 15:57:02.988 I/MonoDroid(25422):   at dotMorten.Xamarin.Forms.NativeAutoSuggestBox+SuggestCompleteAdapter+SuggestFilter.PerformFiltering (Java.Lang.ICharSequence constraint) [0x0001c] in E:\GitHub\dotMorten\XamarinFormsControls\AutoSuggestBox\dotMorten.Xamarin.Forms.AutoSuggestBox\NativeAutoSuggestBox.Android.cs:191 
01-19 15:57:02.988 I/MonoDroid(25422):   at Android.Widget.Filter.n_PerformFiltering_Ljava_lang_CharSequence_ (System.IntPtr jnienv, System.IntPtr native__this, System.IntPtr native_constraint) [0x00011] in <6f8818d6ffa14f4bad6fb5f5d0f0e665>:0 
01-19 15:57:02.988 I/MonoDroid(25422):   at (wrapper dynamic-method) System.Object.61(intptr,intptr,intptr)
```

### Steps to Reproduce

1. Go on a page with the AutoSuggestBox
2. Start typing and select a suggested element (this is important, if you just type something without using the suggestion there is no error)
3. Go on the previous page (or I think, do whatever that would makes the page where the AutoSuggestBox was to be unreferenced)

And thanks for the good work :)

**Comment by dotMorten** [comment]
@Sim-github I wasn't able to repro based on the description above, but I think from the stacktrace I can see what's going on, and I just pushed a speculative fix. Could you try it with the latest source code and confirm that fixes it? (If so I'll push out a v0.6 update)

**Comment by Sim-github** [comment]
Created a test where I reproduce.
https://github.com/Sim-github/AutoSuggestTest

It is a bit more convoluted than I expected, maybe what I'm doing is not the supposed way to manage carousel navigation...
What I do is that I have a CarouselPage where I add my pages, and on some events I remove pages. Here in the example I created, click on the first two buttons (each add a new page in the carousel and navigate to it), on the third page select the word "Test" from the AutoSuggestBox. Then slide back to the first page and click again on the button. What this button does is remove the all next pages in the carousel and add the second page. Somehow doing this raise the error.

(But do this again without selecting "Test" and entering something else works fine... If you manage to figure out what is going on I'm curious...)

**Comment by dotMorten** [comment]
Thank you for the reproducer - That's much appreciated. Unfortunately when running on my Android phone I'm not seeing any issue with the above steps. The app seems to run just fine?!?

**Comment by Sim-github** [comment]
Hum... I only used it in debug with the Android emulator with Visual Studio, could it be that it only happens in the emulator somehow? I will try it with a real phone as soon as I have some time.

**Comment by Sim-github** [comment]
I confirm it works fine on my phone, but always fails in debug mode on the Android emulator... So... do as you wish I guess ^^

**Comment by Sim-github** [comment]
The one that comes with Visual Studio, with version 8.1 Oreo (API 27)
I could try to upgrade to 9.0 to see if that changes something

**Comment by dotMorten** [comment]
The new one provided by Google with the hyper-v support Microsoft added, or the older now deprecated visual studio emulator?

**Comment by Sim-github** [comment]
I think the google one (I didn't even knew there was an older one, but I reinstalled VS 2017 recently so unless the old one is still the default it should be the new one)

**Comment by dotMorten** [comment]
Fixed in v0.6.0
